pkg/storegateway/series_refs.go

```diff
@@ -1171,9 +1172,8 @@ func (s *loadingSeriesChunkRefsSetIterator) multiLookupStringify(symbolizedSet s
 	// This can be released by the caller because loadingSeriesChunkRefsSetIterator doesn't retain it after Next() is called again.
 	set := newSeriesChunkRefsSet(len(symbolizedSet.series), true)
 
-	labelsBuilder := labels.NewScratchBuilder(16)
 	for _, series := range symbolizedSet.series {
-		lset, err := s.indexr.LookupLabelsSymbols(s.ctx, series.lset, &labelsBuilder)
```

Comment:
some of the symbols returned from `LookupLabelsSymbols` are already interned by `StreamBinaryReader.LookupSymbol`. This means we might be duplicating the memory now with `labels_dedupelabels.go`, right? 

Would be nice to get a run of `BenchmarkOpenBlockSeriesChunkRefsSetsIterator`, `BenchmarkBucketStore_Series` and `BenchmarkBucketStore_Series_WithSkipChunks` before we enable it.

Reply:
This PR is preparatory - we are some way from adding `-tags dedupelabels` - but yes there are a few places where the current code builds a map of label strings, and the two maps should be merged into one.
[Here's another](https://github.com/prometheus/prometheus/blob/edd558884b246487d84bee69bb39f0d6897966b3/tsdb/index/index.go#L1123).